File: app/src/main/java/org/illegaller/ratabb/hishoot2i/di/module/AppModule.java
package org.illegaller.ratabb.hishoot2i.di.module;

import org.illegaller.ratabb.hishoot2i.model.tray.TrayManager;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {
    private final Application app;

    public AppModule(Application app) {
        this.app = app;
    }

    @Provides @Singleton Application provideApplication() {
        return app;
    }

    @Provides @Singleton TrayManager provideTrayManager() {
        return new TrayManager(app);
    }
}

File: app/src/main/java/org/illegaller/ratabb/hishoot2i/AppModule.kt
package org.illegaller.ratabb.hishoot2i

import android.app.Application
import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
object AppModule {
    @Provides
    @Singleton
    @JvmStatic
    fun provideApplication(application: Application): Context = application
}